chips/sam4l/src/chip.rs

```diff
@@ -152,4 +153,16 @@ impl Chip for Sam4l {
     fn systick(&self) -> &cortexm4::systick::SysTick {
         self.systick
     }
+
+    fn prepare_for_sleep(&self) {
+        if pm::deep_sleep_ready() {
+            unsafe {
+                cortexm4::scb::set_sleepdeep();
+            }
+        } else {
+            unsafe {
+                cortexm4::scb::unset_sleepdeep();
```

Comment:
This line unsets the "deep sleep" bit, making sure the ARM core doesn't go into deep sleep next time we call `wfi`.